libs/contrib/Data/Vect/Views/Extra.idr

```diff
@@ -0,0 +1,66 @@
+||| Additional views for Vect
+module Data.Vect.Views.Extra
+
+import Control.WellFounded
+import Data.Vect
+import Data.Nat
+
+
+||| View for splitting a vector in half, non-recursively
+public export
+data Split : Vect n a -> Type where
+    SplitNil : Split []
+    SplitOne : Split [x]
+    ||| two non-empty parts
+    SplitPair : {n : Nat} -> {m : Nat} ->
+                {x, y : a} -> {xs : Vect n a} -> {ys : Vect m a} ->
+                Split (x :: xs ++ y :: ys)
+
+total
+splitHelp : {n : Nat} -> (head : a) -> (zs : Vect n a) ->
+            Nat -> Split (head :: zs)
+splitHelp head [] _ = SplitOne
+splitHelp head (z :: zs) 0 = SplitPair {xs = []}
+splitHelp head (z :: zs) 1 = SplitPair {xs = []}
+splitHelp head (z :: zs) (S (S k))
+  = case splitHelp head zs k of
```

Comment:
This recursive call should probably be `splitHelp z zs k`.
This becomes evident when `SplitPair`'s `x, y, xs, ys` are all explicit arguments.

Reply:
Actually, you are right, but the code surprisingly worked correctly.